src/is.test.ts

```diff
@@ -33,16 +33,319 @@ import {
+  });
+
+  it('Should return false when "value" is not Infinity', () => {
+    expect(isInfinity('This is a string')).to.equal(false);
+    expect(isInfinity(100)).to.equal(false);
+    expect(isInfinity(undefined)).to.equal(false);
+    expect(isInfinity(null)).to.equal(false);
+    expect(isInfinity(true)).to.equal(false);
+    expect(isInfinity(false)).to.equal(false);
+    expect(isInfinity([])).to.equal(false);
+    expect(isInfinity({})).to.equal(false);
+    expect(isInfinity()).to.equal(false);
+  });
+});
+
+// isInteger(value)
+describe('isInteger(value)', () => {
+  it('Should be a function', () => {
+    expect(isInteger).to.be.a('function');
+  });
+
+  it('Should return true when "value" is an Integer', () => {
+    expect(isInteger(0)).to.equal(true);
+    expect(isInteger(100)).to.equal(true);
+    expect(isInteger(100.00)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not an Integer', () => {
+    expect(isInteger('This is a String')).to.equal(false);
+    expect(isInteger(100.11)).to.equal(false);
+    expect(isInteger(undefined)).to.equal(false);
+    expect(isInteger(null)).to.equal(false);
+    expect(isInteger(true)).to.equal(false);
+    expect(isInteger(false)).to.equal(false);
+    expect(isInteger([])).to.equal(false);
+    expect(isInteger({})).to.equal(false);
+    expect(isInteger()).to.equal(false);
+  });
+});
+
+// isIntegerArray(value)
+describe('isIntegerArray(value)', () => {
+  it('Should be a function', () => {
+    expect(isIntegerArray).to.be.a('function');
+  });
+
+  it('Should return true when "value" is type number[] and each array entry is an integer', () => {
+    expect(isIntegerArray([0, 0, 0])).to.equal(true);
+    expect(isIntegerArray([1, 2, 3])).to.equal(true);
+    expect(isIntegerArray([0.00, 1.00, 2.00, 3.00])).to.equal(true);
+    expect(isIntegerArray([])).to.equal(true);
+  });
+
+  it('Should return false when "value" is not type number[]', () => {
+    expect(isIntegerArray('This is a String')).to.equal(false);
+    expect(isIntegerArray(100)).to.equal(false);
+    expect(isIntegerArray(undefined)).to.equal(false);
+    expect(isIntegerArray(null)).to.equal(false);
+    expect(isIntegerArray(true)).to.equal(false);
+    expect(isIntegerArray(false)).to.equal(false);
+    expect(isIntegerArray(['String', 2, false])).to.equal(false);
+    expect(isIntegerArray({})).to.equal(false);
+    expect(isIntegerArray()).to.equal(false);
+  });
+
+  it('Should return false when "value" is type number[] but not every array entry is an integer', () => {
+    expect(isIntegerArray([1.1, 2.2, 3.3])).to.equal(false);
+    expect(isIntegerArray([1, 2.2, 3.3])).to.equal(false);
+    expect(isIntegerArray([1, 2, 3.3])).to.equal(false);
+  });
+});
+
+// isNaN(value)
+describe('isNaN(value)', () => {
+  it('Should be a function', () => {
+    expect(isNaN).to.be.a('function');
+  });
+
+  it('Should return true when "value" is NaN', () => {
+    expect(isNaN(NaN)).to.equal(true);
+  });
```

Comment:
Check other ways to get NaN:

```suggestion
  it('Should return true when "value" is NaN', () => {
    expect(isNaN(NaN)).to.equal(true);
    expect(parseInt('No number')).to.equal(true);
    expect(parseFloat('No number')).to.equal(true);
  });
```

Reply:
should I try other ways of NaN? A google search shows that 1*undefined also results in NaN. I am seeing some examples of parseInt like ```parseInt('Joker', 10); // => NaN``` . I was thinking to do that too.

Reply:
I guess 'No number' is a String and it should cover string as an input that is parsed and becomes NaN

Reply:
Right, that was my intention behind parsing `'No number'`. Beyond that, we would just be testing the parse functions :smile: